docs/xxcheckpointgcthread.md

```diff
@@ -0,0 +1,51 @@
+<!--
+* Copyright (c) 2017, 2023 IBM Corp. and others
+*
+* This program and the accompanying materials are made
+* available under the terms of the Eclipse Public License 2.0
+* which accompanies this distribution and is available at
+* https://www.eclipse.org/legal/epl-2.0/ or the Apache
+* License, Version 2.0 which accompanies this distribution and
+* is available at https://www.apache.org/licenses/LICENSE-2.0.
+*
+* This Source Code may also be made available under the
+* following Secondary Licenses when the conditions for such
+* availability set forth in the Eclipse Public License, v. 2.0
+* are satisfied: GNU General Public License, version 2 with
+* the GNU Classpath Exception [1] and GNU General Public
+* License, version 2 with the OpenJDK Assembly Exception [2].
+*
+* [1] https://www.gnu.org/software/classpath/license.html
+* [2] https://openjdk.org/legal/assembly-exception.html
+*
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH
+* Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+# -XX:CheckpointGCThreads
+
+**(Linux&reg; x86, Linux on POWER&reg; (Little Endian), and Linux on IBM Z&reg; only)**
+
+This option restricts the number of threads that the garbage collector uses for parallel operations at the time of taking checkpoint.
+
+:fontawesome-solid-triangle-exclamation:{: .warn aria-hidden="true"} **Restrictions:** This option takes effect only when `-XX:+EnableCRIUSupport` is enabled.
+
+## Syntax
+
+        -XX:CheckpointGCThreads=<number>
+
+| Setting               | Value                            | Default                                                  |
+|-----------------------|-----------------------------------|:---------------------------------------------------------:|
+| `<number>` | Greater than 0 and less than or equal to the number of garbage collection (GC) threads at the VM startup |  4           |
+
+You can specify the number of GC threads at the startup with the [`-Xgcthreads`](xgcthreads.md) option.
+
+## Explanation
+
+When taking a checkpoint, the VM reduces the number of GC threads that was specified at the startup and makes it equal to the thread count specified in the `-XX:CheckpointGCThreads` option. Since restoring threads adds latency to the overall VM restore time, you can minimize the restore time by reducing the GC thread count. After restoring from the checkpoint, the VM increases the GC thread count back to the number of GC threads that the VM had at startup.
```

Comment:
After restoring from the checkpoint, the number of GC thread count is increased based on
- What the JVM decides as default (see explanation under [-Xgcthreads ](https://eclipse-openj9.github.io/openj9-docs-staging/1091/xgcthreads/) and what Tobi https://github.com/eclipse-openj9/openj9-docs/issues/1053#issuecomment-1507613074 ) 
- Or what the user specifies at restore time using `-Xgcthreads`

Reply:
Both `#1` and `#2` are correct. In terms of `#3`,  it is true that the VM increases the GC thread count after restoring, however this specific part is not correct:
> back to the number of GC threads that the VM had at startup (as explained in point 1 in this comment)

This flow chart may help in understanding what happens at restore time:

<img width="409" alt="image" src="https://user-images.githubusercontent.com/9648788/236989778-f536e0ed-79d7-4f02-8afd-39f9e135b31c.png">

Reply:
VM  determines a "restore GC thread count", which is either restore option`-Xgcthreads` or a new default determined by VM (like in `#1`). If `-Xgcthreads` at restore is not specified or invalid (less than checkpoint), then the VM determines a default restore thread count and checks if it's valid (greater than checkpoint).

Reply:
>So that would mean that the VM will then increase the GC thread count back to the number of GC threads that the VM had at startup?

Startup thread count is never considered at restore time.

Reply:
Essentially, after startup, GC Threads can only:
-  At Checkpoint: Shutdown, threads will never be started (not possible for  checkpoint thread count to increase from startup)
-  Restore: Started, threads will never be shutdown (not possible for restore thread count to decsrease from checkpoint)

Reply:
Not relevant for this documentation... The restore behaviour was slightly modified post 0.38 release, this will be reverted back to whats been described above.

Reply:
@RSalman - Could you please confirm that modified content is now ok. Thanks!